### PR TITLE
Add ignoreFunctions: [] to length-zero-no-unit

### PR DESCRIPTION
**Pull request by JuanFML**
Added condition to ignore when function is detected and a test to check it with.

> Which issue, if any, is this issue related to?

Fixes issue #4945 

> Is there anything in the PR that needs further explanation?

I left some comments in the code as it is another type of solution, this is my first PR, so please let me know if the solution wasn't what you were looking for, in order to fix it. There would be still left to add to the README.md file the new option added, although I wanted to get it checked first before adding it.

### CODE REVIEW

File: lib/rules/length-zero-no-unit/index.js [hunk 1]
```diff
@@ -16,16 +16,29 @@ const ruleMessages = require('../../utils/ruleMessages');
 const styleSearch = require('style-search');
 const validateOptions = require('../../utils/validateOptions');
 const valueParser = require('postcss-value-parser');
+//const { truncate } = require('lodash');
```
Comment (vankop) [inline lib/rules/length-zero-no-unit/index.js:19]: could you please delete unneeded comments

File: lib/rules/length-zero-no-unit/index.js [hunk 2]
```diff
@@ -83,13 +96,36 @@ function rule(actual, secondary, context) {
...
+			// });
+
 			const fixPositions = [];
 
 			styleSearch({ source: value, target: '0' }, (match) => {
+				const ignoreFunctions = (options && options.ignoreFunctions) || [];
+
+				if (
+					ignoreFunctions.length > 0 &&
+					ignoreFunctions.map((functionName) => {
```
[the remark this answers is not in the document]
Comment (vankop) [reply lib/rules/length-zero-no-unit/index.js:120]: could you also add another rejected test case, where 
```
{
	code: '.foo { margin-top: var(--y, 0px); margin-bottom: --custom1(0px); };',
	description: 'Should error value in --custom1',
}
```

**Review by vankop — COMMENTED** [review]
Thanks for your PR, some comments:

File: lib/rules/length-zero-no-unit/index.js (hunk 1 above)
Comment (JuanFML) [reply lib/rules/length-zero-no-unit/index.js]: I have made the changes

**Comment by vankop** [comment]
@JuanFML I have reworked things a little bit, take a look

**Comment by JuanFML** [comment]
@vankop Yeah they work perfectly your changes, I don't have to commit again do I? With your commit is all done?

**Comment by vankop** [comment]
> I don't have to commit again do I?
yes, all is fine

**Review by vankop — APPROVED** [review]

**Review by hudochenkov — COMMENTED** [review]
Thank you for working on this!

Could you add test cases, please?

```css
/* it should report only first 0px */
a { margin: 0px var(--prop, 0px) }

/* it should report only second 0px */
a { margin: calc(5px + var(--prop, 0px)) 0px }
```

**Comment by jeddy3** [comment]
Revisiting after the refactoring in https://github.com/stylelint/stylelint/pull/5256.

I think a simple conditional just after [line 52](https://github.com/stylelint/stylelint/blob/master/lib/rules/length-zero-no-unit/index.js#L52) should work now:

```js
if (isFunction(valueNode) && optionsMatches(secondary, 'ignoreFunctions', value)) return false;
```

With the following tests:

```js
testRule({
	ruleName,
	config: [true, { ignoreFunctions: ['var', /^--/] }],
	accept: [
		{
			code: 'a { top: var(--foo, 0px); }',
		},
		{
			code: 'a { top: --bar(0px); }',
		},
	],
	reject: [
		{
			code: 'a { margin: var(--foo, 0px) 0px --bar(0px); }',
			fixed: 'a { margin: var(--foo, 0px) 0 --bar(0px); }',

			message: messages.rejected,
			line: 1,
			column: 30,
		},
	],
});
```

We'll need to require `lodash` on line 5:

```js
const _ = require('lodash');
```

And update the options validator on line 41:

```js
ignoreFunctions: [_.isString, _.isRegExp],
```

And add a `isFunction` function on line 154:

```js
function isFunction({ type }) {
	return type === 'function';
}
```

The docs can look like [this](https://github.com/stylelint/stylelint/tree/72d7bab805af3b7a02df5081b83f007b3b290af0/lib/rules/property-no-vendor-prefix#optional-secondary-options), e.g.

Given:

```
["var", "/^--/"]
```

The following patterns are _not_ considered violations:


```css
a { top: var(--foo, 0px); }
```


```css
a { top: --bar(0px); }
```

@JuanFML Would you like to update the pull request?

**Pull request closed without merging**